exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/limit/TestLimitWithExchanges.java

```diff
@@ -26,4 +30,34 @@
   public void testLimitWithExchanges() throws Exception{
     testPhysicalFromFile("limit/limit_exchanges.json");
   }
+
+  @Test
+  public void testPushLimitPastUnionExchange() throws Exception {
```

Comment:
I think we should add a simple test with data validation for the LIMIT pushdown.  Also include OFFSET + LIMIT.